controllers/hetznercluster_controller_test.go

```diff
@@ -840,3 +842,100 @@ var _ = Describe("reconcileRateLimit", func() {
 		Expect(reconcileRateLimit(hetznerCluster, testEnv.RateLimitWaitTime)).To(BeFalse())
 	})
 })
+
+func TestSetControlPlaneEndpoint(t *testing.T) {
+	t.Run("function should return false if hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled is not enabled and hetznerCluster.Spec.ControlPlaneEndpoint is nil", func(t *testing.T) {
+		hetznerCluster := &infrav1.HetznerCluster{Spec: infrav1.HetznerClusterSpec{ControlPlaneLoadBalancer: infrav1.LoadBalancerSpec{Enabled: false}, ControlPlaneEndpoint: nil}}
+		got := SetControlPlaneEndpoint(hetznerCluster)
+		if (hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled == true){
```

Comment:
we never change anything in ControlPlaneLoadBalancer. You really don't have to test it. Focus on changes in ControlPlaneEndpoint. You can remove this